Updating to v5

SocksTheWolf opened the pull request:
Also sort of updates the websocket messages to the latest format that Tiltify uses (I use the polling mechanism as websocket had too many issues and was out of scope to fix that). Tested very loosely with my bridge program, but it works for me.

Closes #1

SocksTheWolf committed:
Quickly update the paths for Tiltify from v3 to v5

Adds in more versions and also the accessibility paths (public vs private).

SocksTheWolf committed:
Update to .net6

For allowing for multi-platform usability at minimum.

SocksTheWolf committed:
Update to latest Modals

SocksTheWolf committed:
Added auth endpoint

also allow for other objects to update settings (such as auth), and enable POST requests

JustDerb commented on ApiAccessLevel.cs:
```diff
@@ -0,0 +1,34 @@
+                    return "/api/private";
+                case ApiAccessLevel.None:
+                default:
+                    return "/";
```
Shouldn't this branch follow similar output to `OAuth` and the `V3` branches?

```suggestion
                    return "";
```

JustDerb commented on WebSocketClient.cs:
```diff
@@ -175,8 +175,6 @@ private Task StartListenerTask()
                             break;
                         case WebSocketMessageType.Binary:
                             break;
-                        default:
-                            throw new ArgumentOutOfRangeException();
```
Any reason this was removed?

SocksTheWolf replied:
I believe I removed this line while I was debugging some client connection issues, but then never added it back. I'll go ahead and put it back.

JustDerb reviewed (changes requested):
Thanks for the changes! I have a couple small suggestions here. I use the websocket in https://github.com/JustDerb/RoR2-VsTwitch so when I upgrade that mod/library I'll see if the push websocket stuff works (instead of the polling).

SocksTheWolf committed:
Rename WebDonationInformation

This should be called WebSocketDonationInformation for clarity.

SocksTheWolf committed:
Feedback changes

SocksTheWolf committed:
Remove nullable warnings

Add in the proper nullable fields to the various properties that were added that can be null.

SocksTheWolf commented:
Branch updated.

I ran into issues with the websocket itself, I'm not entirely sure why. I was getting updates for when donations were coming in for a few minutes but Tiltify would suddenly stop sending anything (as if the connection was reset silently [though I observed no change in socket state]).

While connected, I did notice that messages + heartbeat were correct, serializing and being responded to (I did get a "phx_reply" with an "ok" status), as Tiltify did update their message format (which was just changing the message prefix/suffix in a few areas as seen in the image below)

![image](https://github.com/user-attachments/assets/a426a016-3fb9-4a3e-b6b5-15c99371fef6)

Time forbid trying to debug the issue further, as much as I would rather a websocket solution (hell even an official EventSub/PubSub like Twitch has) over polling (especially since the limits and such for their polling system are not well documented publicly).

SocksTheWolf committed:
Swap to nullable restore

This way we don't step on anyone's toes.

SocksTheWolf reviewed (commented):
changes implemented

SocksTheWolf committed:
Add team campaign support

JustDerb commented:
>  I was getting updates for when donations were coming in for a few minutes but Tiltify would suddenly stop sending anything (as if the connection was reset silently [though I observed no change in socket state]).

I can repro with this branch as well. For some reason on my side it stops after a donation goes through (no more heartbeat pings happen).

I'm adding some more debugging to track things down, but haven't come up with anything yet.

Also, you can add another logger into https://github.com/JustDerb/Tiltify-Client/blob/4636d8c5a9fd854b9eddafaf29a234a887c63a0a/TiltifyWebSocket.cs#L84 to get more details. I might just simplify the logging for this class and just have one (there was two for data parsing and other logging).
